src/Symfony/Component/Messenger/Tests/Transport/Serialization/SerializerTest.php

```diff
@@ -207,7 +207,40 @@ public function testEncodedSkipsNonEncodeableStamps()
         $encoded = $serializer->encode($envelope);
         $this->assertStringNotContainsString('DummySymfonySerializerNonSendableStamp', print_r($encoded['headers'], true));
     }
+
+    public function testDecodingFailedConstructorDeserialization()
+    {
+        $this->expectException(MessageDecodingFailedException::class);
+
+        $serializer = new Serializer();
+
+        $serializer->decode([
+            'body' => '{}',
+            'headers' => ['type' => DummySymfonySerializerInvalidConstructor::class],
+        ]);
+    }
+
+    public function testDecodingStampFailedDeserialization()
+    {
+        $this->expectException(MessageDecodingFailedException::class);
```

Comment:
Move this down one line. 

This test is used to prove that `MessageDecodingFailedException` is thrown when running `$serializer->decode`. But currently, I can no be sure if the exception is thrown at `$serializer = new Serializer();` or `$serializer->decode`.

Reply:
I've done this for my changes.
I noticed this is the case for other tests as well in the same file. I left those as-is for now.
Let me know if I should update these as well.

Reply:
Thank you. 
One should never do unrelated changes in a PR. That will make reviews more difficult and possibly delay the merge of the feature you intended. 

Feel free to submit a new PR with those changes and we can discuss them separately.